WcaOnRails/app/models/competition.rb

```diff
@@ -509,12 +524,20 @@ def end_date
   end
 
   private def events_must_be_valid
-    invalid_events = events - Event.all_official - Event.all_deprecated
+    invalid_events = events - Event.official - Event.deprecated
```

Comment:
Perhaps this validation is not needed now that Events is a proper table and association?